### PR TITLE
Rename kremlin package into karamel

### PR DESCRIPTION
**Pull request by R1kM**
Following instructions at https://opam.ocaml.org/doc/Tricks.html#How-to-rename-a-package, rename the previous kremlin package into karamel.

Fixes https://github.com/FStarLang/karamel/issues/300

Ping @msprotz

### CODE REVIEW

File: packages/karamel/karamel.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "process"
+  "fix"
+  "visitors"
+  "wasm" {= "1.1.1"}
+  "ppx_deriving"
+  "ppx_deriving_yojson"
+  "fstar" {= "2022.01.15"}
+]
+depexts: ["coreutils"] {os = "macos" & os-distribution = "homebrew"}
+conflicts: "kremlin" {!= "transition"}
```
Comment (kit-ty-kate) [inline packages/karamel/karamel.1.0.0/opam:27]: this is confusing. transition package usually simply depend on the new package for easy transition but here this makes it impossible to do. What’s the purpose of the transition package then?

Comment (kit-ty-kate) [reply packages/karamel/karamel.1.0.0/opam:27]: nevermind i read it backward, now that the transition package depends on the right package it's perfectly fine. Sorry for the dumb comment.

**Comment by kit-ty-kate** [comment]
Thanks

File: packages/karamel/karamel.1.0.0/opam (hunk 1 above)
Comment (R1kM) [reply packages/karamel/karamel.1.0.0/opam:27]: Apologies, I might have misunderstood the documentation. Thanks for the help and for merging!

Comment (kit-ty-kate) [reply packages/karamel/karamel.1.0.0/opam:27]: Sorry if something was badly explained. If you think something can be done better feel free to open a PR to modify this documentation: https://github.com/ocaml/opam/blame/master/doc/pages/Tricks.md

**Pull request merged**